### PR TITLE
Fix empty array copy bug.

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

**Comment by mijovic** [comment]
This was a problem with `memory->storage` only?

**Comment by chriseth** [comment]
I can add more tests for copying from calldata and from storage.

**Comment by mijovic** [comment]
> I can add more tests for copying from calldata and from storage.

Would be great. Otherwise, looks good

**Review by mijovic — DISMISSED** [review]

**Comment by chriseth** [comment]
This is still missing bug list entry, changelog entry, blog post.

**Comment by mijovic** [comment]
> This is still missing bug list entry, changelog entry, blog post.

Do we need to wait for blog post to be finished before merging this?

**Comment by chriseth** [comment]
no need to wait for the blog post

File: test/libsolidity/semanticTests/array/copying/empty_bytes_copy.sol [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        assembly  { sstore(otherData.slot, 0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff00) }
+        data = otherData;
+        data.push();
+        return data[0];
+    }
+}
+// ----
+// fromMemory() -> 0x00
+// fromCalldata(bytes): 0x40, 0x60, 0x00, 0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff -> 0x00
+// fromStorage() -> 0x00
```
Comment (mijovic) [inline test/libsolidity/semanticTests/array/copying/empty_bytes_copy.sol:28]: Without changes, these two are failing as well. Maybe this needs to be mentioned in bug list and changelog, as it is not only the case when we copy from memory?

Comment (chriseth) [reply test/libsolidity/semanticTests/array/copying/empty_bytes_copy.sol:28]: The storage one is also failing? Hm...

**Review by mijovic — APPROVED** [review]

**Pull request merged**